diff --git a/package/musl/Config.in b/package/musl/Config.in
index d9ab30ea92..927c47ea4f 100644
--- a/package/musl/Config.in
+++ b/package/musl/Config.in
@@ -14,7 +14,7 @@ config BR2_PACKAGE_MUSL_ARCH_SUPPORTS
 	default y if BR2_powerpc
 	default y if BR2_powerpc64
 	default y if BR2_powerpc64le
-	default y if BR2_RISCV_64
+	default y if BR2_RISCV_32 || BR2_RISCV_64
 	default y if BR2_s390x
 	default y if BR2_sh
 	default y if BR2_x86_64
diff --git a/package/musl/musl.hash b/package/musl/musl.hash
index 8ec5c241e9..a8a3b307d5 100644
--- a/package/musl/musl.hash
+++ b/package/musl/musl.hash
@@ -1,4 +1,4 @@
 # Locally calculated after checking pgp signature from
-# http://musl.libc.org/releases/musl-1.2.4.tar.gz.asc
-sha256  7a35eae33d5372a7c0da1188de798726f68825513b7ae3ebe97aaaa52114f039  musl-1.2.4.tar.gz
+# http://musl.libc.org/releases/musl-1.2.5.tar.gz.asc
+sha256  a9a118bbe84d8764da0ea0d28b3ab3fae8477fc7e4085d90102b8596fc7c75e4  musl-1.2.5.tar.gz
 sha256  f9bc4423732350eb0b3f7ed7e91d530298476f8fec0c6c427a1c04ade22655af  COPYRIGHT
diff --git a/package/musl/musl.mk b/package/musl/musl.mk
index 9976e39950..8df4163e09 100644
--- a/package/musl/musl.mk
+++ b/package/musl/musl.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-MUSL_VERSION = 1.2.4
+MUSL_VERSION = 1.2.5
 MUSL_SITE = http://musl.libc.org/releases
 MUSL_LICENSE = MIT
 MUSL_LICENSE_FILES = COPYRIGHT
